packages/client/src/arpa_reporter/views/NewUpload.vue

```diff
@@ -1,111 +1,164 @@
+        },
+        {
+          type: 'textarea',
+          label: 'Notes',
+          name: 'notes',
+        },
+        {
+          component: 'div',
+          class: 'row',
+          children: [
+            {
+              name: 'submit',
+              type: 'submit',
+              label: 'Submit',
+              inputClass: ['btn', 'btn-primary'],
+            },
+            {
+              type: 'button',
+              label: 'Reset',
+              inputClass: ['btn', 'btn-danger', 'ml-2'],
+              '@click': this.onReset,
+            },
+          ],
+        },
+      ],
     };
   },
-  computed: {
-    uploadButtonLabel() {
-      return this.uploading ? 'Uploading...' : 'Upload';
-    },
-    uploadDisabled() {
-      return this.files === null || this.uploading;
-    },
+  created() {
+    this.values = this.initialValues();
   },
   methods: {
-    changeFiles(evt) {
-      this.files = evt.target.files;
+    reportingPeriodSelectItems() {
+      // create an object from this.$store.getters.viewableReportingPeriods where every key is period.id and every value is period.name
+      return Object.fromEntries(this.$store.getters.viewableReportingPeriods.map((period) => [period.id, period.name]));
     },
-    async onSubmit() {
-      const file = this.files[0];
-
-      if (!file) {
-        this.$refs.files.focus();
-        return;
-      }
-
-      this.error = null;
-      this.uploading = true;
-
-      const formData = new FormData();
-      formData.append('spreadsheet', file);
-
-      // TODO: (#896) Actually include data when submitting
-      // formData.append('notes', '<b>TEST & STRING!</b>');
-
+    agencySelectItems() {
+      return Object.fromEntries(this.$store.state.agencies.map((agency) => [agency.id, agency.name]));
+    },
+    // eslint-disable-next-line no-unused-vars
+    async uploadFile(file, progress, error, options) {
+      /* VueFormulate traditionally calls this method to upload the file
+         to some remote location, but for now, we don't want to do that.
+         It may eventually make sense to do that when we don't have access
+         to the file system, e.g in a lambda context. For now though,
+         we just return the file object. */
+      return file;
+    },
+    async onSubmit(formData) {
+      // This is technically a FileList, allowing for multiple uploads if configured
+      // this way, but for now we have only a single file.
+      const file = formData.spreadsheet[0];
+      const uploadData = {
+        reportingPeriodId: formData.reportingPeriodId,
+        agencyId: formData.agencyId,
+        expenditure_code: formData.expenditure_code,
+        notes: formData.notes,
+      };
+      const uploadFormData = new FormData();
+      uploadFormData.append('spreadsheet', file);
+      Object.keys(uploadData).forEach((key) => {
+        const value = uploadData[key];
+        if (value) {
+          uploadFormData.append(key, value);
+        }
+      });
```

Comment:
```suggestion
      const uploadFormData = new FormData();
      uploadFormData.append('spreadsheet', data.spreadsheet[0]);
      Object.entries(_.omit(data, ['spreadsheet'])).forEach(([key, value]) => {
        if (value) {
          uploadFormData.append(key, value);
        }
      });
```

Nit: We do have lodash available in this project.  Using that, we can probably simplify this block a little!

Reply:
Ooh, I have less familiarity with all the functionality available in lodash - nice to have this!

Reply:
I have found the lodash docs to be pretty good!  I do still have to check them pretty frequently to double-check the exact syntax.

https://lodash.com/docs/